### PR TITLE
expression: open push down switcher for part of CAST functions

### PR DESCRIPTION
**Pull request by lonng**
Signed-off-by: Lonng <heng@lonng.org>

### What problem does this PR solve? 

Open the CAST push-down switcher and fix some execution plans.

### What is changed and how it works?

We have fixed all CAST functions related inconsistent between TiDB and TiKV, it's time to open the switcher default.

### Check List 

Tests 

 - Unit test
 - Integration test

Release note

 - No need

### CODE REVIEW

**Comment by iosmanthus** [comment]
`copr-test` may fail, more detail on: https://github.com/pingcap/tidb/issues/14280

File: util/codec/codec.go [hunk 1]
```diff
@@ -984,10 +984,19 @@ func (decoder *Decoder) DecodeOne(b []byte, colIdx int, ft *types.FieldType) (re
 		chk.AppendBytes(colIdx, v)
 	case decimalFlag:
 		var dec *types.MyDecimal
-		b, dec, _, _, err = DecodeDecimal(b)
+		var frac int
+		b, dec, _, frac, err = DecodeDecimal(b)
 		if err != nil {
 			return nil, errors.Trace(err)
 		}
+		if ft.Decimal != types.UnspecifiedLength && frac > ft.Decimal {
```
Comment (XuHuaiyu) [inline util/codec/codec.go]: Is this check covered by any test cases?

Comment (lonng) [reply util/codec/codec.go]: Yes, this BUG is triggered by test cases after opening CAST swither.

**Pull request merged**